src/wazuh_modules/wm_sca.c

```diff
@@ -824,8 +852,13 @@ static int wm_sca_do_scan(OSList *p_list,cJSON *profile_check,OSStore *vars,wm_s
                 nbuf = p_check->valuestring;
                 mdebug2("Rule is: %s",nbuf);
 
+                /* Make a copy of the rule */
+                char *rule_cp;
+                os_strdup(nbuf, rule_cp);
```

Comment:
It is necessary to copy the `nbuf` variable for processing the rules?

Reply:
Yes, it is necessary becasue the function `wm_sca_get_value` crops the string received as argument, so the rule is lost.

Reply:
Ok, makes sense. Thank you!